### PR TITLE
[Logs] New page for Logpush to HTTP destination.

### PR DESCRIPTION
**Pull request by angelampcosta**

### CODE REVIEW

File: content/logs/get-started/http-destination.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+pcx-content-type: concept
+title: Enable HTTP destination 
+weight: 50
+---
+
+# Enable HTTP destination
+
+Cloudflare Logpush now supports the ability to send logs to arbitrary HTTP endpoints.
+
+Note that when using Logpush to HTTP endpoints, Cloudflare customers are expected to perform their own authentication of the pushed logs. For example, customers may specify a secret token in the URL or an HTTP header of the Logpush destination.
```
Comment (deadlypants1973) [inline content/logs/get-started/http-destination.md]: Small thing but does the customer need to specify a secret token to complete the task successfully? If so, then changing 'may' to 'may need to' would be necessary.

File: content/logs/get-started/http-destination.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## Manage via API
+
+To create a job, make a `POST` request to the [Logpush job creation endpoint URL](/logs/reference/logpush-api-configuration/) with the appropriate parameters.
```
Comment (deadlypants1973) [inline content/logs/get-started/http-destination.md]: Never heard of 'creating a job' before, is it another way to say logpush? Job wasn't defined before this section.

File: content/logs/get-started/http-destination.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+The supported parameters are as follows:
+
+- Fields that are unchanged from other sources:
+    - **dataset** (required): For example, `http_requests`.
+    - **name** (optional): We suggest using your domain name as the job name.
+    - **logpull_options** (optional): Refer to [Logpush API configuration options](/logs/reference/logpush-api-configuration/#options) to configure fields, sample rate, and timestamp format.
+- Unique fields: 
+    - **destination_conf**: Where to send the logs. This consists of an endpoint url and HTTP headers used.
```
Comment (deadlypants1973) [inline content/logs/get-started/http-destination.md]:
```suggestion
    - **destination_conf**: Where to send the logs. This consists of an endpoint URL and HTTP headers used.
```

File: content/logs/get-started/http-destination.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        - Any `"header_*"` URL parameters will be taken to set request headers.
+            - Thus, the HTTPS endpoint cannot have custom URL parameters that conflicts with this.
+            - These parameters must be properly URL-encoded (that is, use `"%20"` for a whitespace), otherwise some special characters may be decoded incorrectly.
+        - destination_conf may have more URL parameters in addition to special `"header_*"` parameters.
```
Comment (deadlypants1973) [inline content/logs/get-started/http-destination.md]:
```suggestion
        - `destination_conf` may have more URL parameters in addition to special `"header_*"` parameters.
```

File: content/logs/get-started/http-destination.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            - Non URL-encoded special chars will be encoded when uploading.
```
Comment (deadlypants1973) [inline content/logs/get-started/http-destination.md]: Characters?

```suggestion
            - Non URL-encoded special characters will be encoded when uploading.
```

File: content/logs/get-started/http-destination.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        - Example: `https://logs.example.com?header_Authorization=Basic%20REDACTED&tags=host:theburritobot.com,dataset:http_requests`
+    - **max_upload_bytes** (optional): The maximum uncompressed file size of a batch of logs. This must be at least 5 MB Note that there is not a way to set a minimum file size; this means that log files may be much smaller than this batch size.
```
Comment (deadlypants1973) [inline content/logs/get-started/http-destination.md]:
```suggestion
    - **max_upload_bytes** (optional): The maximum uncompressed file size of a batch of logs. This must be at least 5 MB. Note that there is no way to set a minimum file size; this means that log files may be much smaller than this batch size.
```

File: content/logs/get-started/http-destination.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+The supported parameters are as follows:
+
+- Fields that are unchanged from other sources:
+    - **dataset** (required): For example, `http_requests`.
+    - **name** (optional): We suggest using your domain name as the job name.
+    - **logpull_options** (optional): Refer to [Logpush API configuration options](/logs/reference/logpush-api-configuration/#options) to configure fields, sample rate, and timestamp format.
+- Unique fields: 
+    - **destination_conf**: Where to send the logs. This consists of an endpoint url and HTTP headers used.
+        - Any `"header_*"` URL parameters will be taken to set request headers.
```
Comment (deadlypants1973) [inline content/logs/get-started/http-destination.md]:
```suggestion
        - Any `"header_*"` URL parameters will be used to set request headers.
```

File: content/logs/get-started/http-destination.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+            - Thus, the HTTPS endpoint cannot have custom URL parameters that conflicts with this.
```
Comment (deadlypants1973) [inline content/logs/get-started/http-destination.md]: Remove 'Thus' as this is a new paragraph and make 'this' into a proper noun (I am guessing that it is the URL parameters?)

```suggestion
            - The HTTPS endpoint cannot have custom URL parameters that conflicts with any `"header_*"` URL parameters you have set.
```

File: content/logs/get-started/http-destination.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+            - These parameters must be properly URL-encoded (that is, use `"%20"` for a whitespace), otherwise some special characters may be decoded incorrectly.
+        - destination_conf may have more URL parameters in addition to special `"header_*"` parameters.
+            - Non URL-encoded special chars will be encoded when uploading.
+        - Example: `https://logs.example.com?header_Authorization=Basic%20REDACTED&tags=host:theburritobot.com,dataset:http_requests`
+    - **max_upload_bytes** (optional): The maximum uncompressed file size of a batch of logs. This must be at least 5 MB Note that there is not a way to set a minimum file size; this means that log files may be much smaller than this batch size.
+    - **max_upload_records** (optional): The maximum number of log lines per batch. This must be at least 1000 lines or more. Note that there is not a way to specify a minimum number of log lines per batch; this means that log files may contain many fewer lines than this.
```
Comment (deadlypants1973) [inline content/logs/get-started/http-destination.md]:
```suggestion
    - **max_upload_records** (optional): The maximum number of log lines per batch. This must be at least 1000 lines or more. Note that there is no way to specify a minimum number of log lines per batch; this means that log files may contain many fewer lines than this.
```

File: content/logs/get-started/http-destination.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+{{<Aside type="note" header="Note">}}
+**Ownership_challenge** is not required.
```
Comment (deadlypants1973) [inline content/logs/get-started/http-destination.md]: Should this be in backticks (`)? Also kind of random, should there be more context here?

File: content/logs/reference/logpush-api-configuration/_index.md [hunk 11]
```diff
@@ -152,8 +152,12 @@ The four options that you can customize are:
 2.  **Sampling rate**: Value can range from `0.001` to `1.0` (inclusive). `sample=0.1` means `return 10% (1 in 10) of all records`.
 3.  **Timestamp format**: The format in which timestamp fields will be returned. Value options: `unixnano` (default), `unix`, `rfc3339`.
 4.  **Optional redaction for CVE-2021-44228**: This option will replace every occurrence of `${` with `x{`.  To enable it, set `CVE-2021-44228=true`.
+5.  **max_upload_bytes** (optional): The maximum uncompressed file size of a batch of logs. This must be at least 5 MB. Note that there is not a way to set a minimum file size; this means that log files may be much smaller than this batch size.
```
Comment (deadlypants1973) [inline content/logs/reference/logpush-api-configuration/_index.md]:
```suggestion
5.  **max_upload_bytes** (optional): The maximum uncompressed file size of a batch of logs. This must be at least 5 MB. Note that there is no way to set a minimum file size; this means that log files may be much smaller than this batch size.
```

File: content/logs/reference/logpush-api-configuration/_index.md [hunk 12]
```diff
@@ -152,8 +152,12 @@ The four options that you can customize are:
...
+6.  **max_upload_records** (optional): The maximum number of log lines per batch. This must be at least 1000 lines or more. Note that there is not a way to specify a minimum number of log lines per batch; this means that log files may contain many fewer lines than this.
```
Comment (deadlypants1973) [inline content/logs/reference/logpush-api-configuration/_index.md]:
```suggestion
6.  **max_upload_records** (optional): The maximum number of log lines per batch. This must be at least 1000 lines or more. Note that there is no way to specify a minimum number of log lines per batch; this means that log files may contain many fewer lines than this.
```

File: content/logs/reference/logpush-api-configuration/_index.md [hunk 13]
```diff
@@ -152,8 +152,12 @@ The four options that you can customize are:
...
 
-**Note**: The **CVE-2021-44228** parameter can only be set through the API at this time. Updating your Logpush job through the UI will set this option to false.
+{{<Aside type="note" header="Note">}}
+The **CVE-2021-44228** parameter can only be set through the API at this time. Updating your Logpush job through the UI will set this option to false.
```
Comment (deadlypants1973) [inline content/logs/reference/logpush-api-configuration/_index.md]:
```suggestion
The **CVE-2021-44228** parameter can only be set through the API at this time. Updating your Logpush job through the dashboard will set this option to false.
```

**Review by deadlypants1973 — APPROVED** [review]

**Review by soheiokamoto — APPROVED** [review]

File: content/logs/get-started/http-destination.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
+---
+pcx-content-type: concept
+title: Enable HTTP destination 
+weight: 50
+---
+
+# Enable HTTP destination
```
Comment (tanushree-sharma) [inline content/logs/get-started/http-destination.md]: Can we move this page under 'Enable Destinations' similar to how our other destinations are nested

File: content/logs/get-started/http-destination.md [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+  "name": "theburritobot.com-https",
+  "logpull_options": "fields=RayID,EdgeStartTimestamp&timestamps=rfc3339",
+  "destination_conf": "https://logs.example.com?header_Authorization=Basic%20REDACTED&tags=host:theburritobot.com,dataset:http_requests",
+  "max_upload_bytes": 5000000,
+  "max_upload_records": 1000,
+  "dataset": "http_requests",
+  "enabled": true
+}' \
+-H "X-Auth-Email: $X_AUTH_EMAIL" \
+-H "X-Auth-Key: $X_AUTH_KEY"
```
Comment (tanushree-sharma) [inline content/logs/get-started/http-destination.md]: Change these to `"X-Auth-Email: <EMAIL>"` and `"X-Auth-Key: <API_KEY>"` for consistency

File: content/logs/get-started/http-destination.md (hunk 1 above)
Comment (tanushree-sharma) [reply content/logs/get-started/http-destination.md]: It's not a requirement we enforce this from our side. I think 'may' actually fits better in this case.

File: content/logs/get-started/http-destination.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
+---
+pcx-content-type: concept
+title: Enable HTTP destination 
+weight: 50
+---
+
+# Enable HTTP destination
+
+Cloudflare Logpush now supports the ability to send logs to arbitrary HTTP endpoints.
```
Comment (jherre) [inline content/logs/get-started/http-destination.md]: I don't think "arbitrary" is exactly the right word for this.  I know what you mean but arbitrary could also be read as random.  How about "configurable"?

**Review by tanushree-sharma — APPROVED** [review]

**Pull request merged**